### PR TITLE
session events: update API

### PR DESCRIPTION
**Pull request by Danielius1922**

### CODE REVIEW

File: include/oc_session_events.h [hunk 1]
```diff
@@ -16,65 +16,65 @@
...
 
-/**
-  @brief Callback function to pass the session event infomation to App.
-  @param endpoint  endpoint info which the session event is happened.
-  @param state  enum values in oc_session_state_t.
-*/
-typedef void (*session_event_handler_t)(const oc_endpoint_t *endpoint,
-                                        oc_session_state_t state);
+#define OC_ERR_SESSION_EVENT_HANDLER_NOT_FOUND -6
+#define OC_ERR_SESSION_EVENT_HANDLER_REMOVED -7
```
Comment (jkralik) [inline include/oc_session_events.h:44]: This `OC_ERR_SESSION_EVENT_HANDLER_NOT_FOUND` never comes from the handler.

And OC_ERR_SESSION_EVENT_HANDLER_REMOVED is used only when networks is shutdown - but it is expected that all managers removed all registered callback. I think we don't need this.

File: include/oc_session_events.h [hunk 2]
```diff
@@ -16,65 +16,65 @@
...
+ *              if < 0 an error value
+ * @param user_data user data provided by the user to
+ * oc_add_session_event_callback_v1
  *
- * @param endpoint stop event on endpoint
+ * @see oc_add_session_event_callback_v1
  */
-void oc_session_end_event(const oc_endpoint_t *endpoint);
+typedef void (*session_event_handler_v1_t)(const oc_endpoint_t *endpoint,
+                                           int state, void *user_data);
```
Comment (jkralik) [inline include/oc_session_events.h:70]: if we more err states than state could be `oc_session_state_t` type

File: api/oc_session_events.c [hunk 3]
```diff
@@ -162,6 +163,167 @@ oc_session_end_event(const oc_endpoint_t *endpoint)
...
+  }
+  if (session_event_remove_callback(oc_session_event_versioned_handler(cb),
+                                    NULL, false) < 0) {
+    return -1;
+  }
+  return 0;
+}
+
+int
+oc_remove_session_event_callback_v1(session_event_handler_v1_t cb,
```
Comment (jkralik) [inline api/oc_session_events.c:287]: return bool (true for success, false for not exist)

File: include/oc_network_monitor.h [hunk 4]
```diff
@@ -21,45 +21,84 @@
...
-*/
+ * @brief Add the callback to receive session event notifications.
+ *
+ * @param cb The callback to be added (cannot be NULL).
+ * @param user_data user data passed to the callback when invoked
+ * @return 0 on success
+ * @return -1 on error
+ */
+OC_API
+int oc_add_session_event_callback_v1(session_event_handler_v1_t cb,
```
Comment (jkralik) [inline include/oc_network_monitor.h:73]: return bool instead of int ?

File: include/oc_network_monitor.h [hunk 5]
```diff
@@ -21,45 +21,84 @@
...
+ oc_add_session_event_callback_v1
+ * @param ignore_user_data ignore user_data and match only the function pointer
+ * @return 0 on success
+ * @return OC_ERR_SESSION_EVENT_HANDLER_NOT_FOUND when no match was found
+ * @return -1 other errors
+ *
+ * @see oc_add_session_event_callback_v1
+ */
+OC_API
+int oc_remove_session_event_callback_v1(session_event_handler_v1_t cb,
```
Comment (jkralik) [inline include/oc_network_monitor.h:99]: return `bool` instead of `int` ?

File: include/oc_network_monitor.h (hunk 4 above)
Comment (Danielius1922) [reply include/oc_network_monitor.h:73]: wanted to keep it consistent with `oc_add_session_event_callback` and `oc_remove_session_event_callback` which return error value.

**Review by jkralik — APPROVED** [review]

**Pull request merged**